lib/active_model/serializer/array_serializer.rb

```diff
@@ -13,7 +14,12 @@ def initialize(objects, options = {})
             :serializer,
             ActiveModel::Serializer.serializer_for(object)
           )
-          serializer_class.new(object, options.except(:serializer))
+
+          if serializer_class.nil?
+            fail NoSerializerError, "No serializer found for object: #{object.inspect}"
+          else
+            serializer_class.new(object, options.except(:serializer))
+          end
```

Comment:
What do you think about having serializer_for return nil for if any elements don't have an associated serializer?  That would maintain current PR behavior, I think, might make more sense SRP-wise, but might be a performance hit.

Should we add a test for a collection or association that contains an element with a known serializer and another with a nil serializer class?  I'm really not sure how to handle that in AMS without more complexity.  I'd rather push that back to the end user, but that could be laziness.